### PR TITLE
Fix No results found iutruyentranh

### LINKED ISSUE #11389: IuTruyenTranh: No results found
### Source information

IuTruyenTranh

### Steps to reproduce

No results were found !

### Expected behavior

Hope u fix it soon
Can u add more vietnamese source :
http://lxhentai.com/
Thks

### Actual behavior

1

### Tachiyomi version

0.13.1

### Android version

12

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by dejavui**
Checklist:
Closes #11389
- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [ ] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt [hunk 1]
```diff
@@ -1,85 +1,238 @@
...
+    private class UserFilter : Filter.Text("Đăng bởi thành viên")
+    private class AuthorFilter : Filter.Text("Tên tác giả")
+    private class SourceFilter : Filter.Text("Nguồn/Nhóm dịch")
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        return when {
+            query.startsWith(PREFIX_ID_SEARCH) -> {
+                val id = query.removePrefix(PREFIX_ID_SEARCH).trim()
+                if (id.isEmpty()) {
+                    throw Exception("ID tìm kiếm không hợp lệ.")
```
Comment (alessandrojean) [inline src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt:164]: It's better to return an `Observable.error` in this case I guess.

File: src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt [hunk 2]
```diff
@@ -1,85 +1,238 @@
...
 
-    override fun popularMangaRequest(page: Int): Request {
-        return GET("$baseUrl/genre/$page?popular", headers)
-    }
-
-    override fun latestUpdatesRequest(page: Int): Request {
-        return GET("$baseUrl/latest/$page", headers)
-    }
+    override fun latestUpdatesRequest(page: Int) = GET(
+        baseUrl.toHttpUrl().newBuilder().apply {
```
Comment (alessandrojean) [inline src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt:48]: No need to build manually the URL here and in some other places, you can use simple string interpolation.

File: src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt [hunk 3]
```diff
@@ -1,85 +1,238 @@
...
         }
-        if (genres.isNotEmpty()) url.addQueryParameter("genres", genres.joinToString(","))
-        if (genresEx.isNotEmpty()) url.addQueryParameter("genres-exclude", genresEx.joinToString(","))
-
-        return GET(url.toString(), headers)
     }
 
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) = GET(
+        baseUrl.toHttpUrl().newBuilder().apply {
+            addPathSegment("search.php")
```
Comment (alessandrojean) [inline src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt:175]: Here you can keep the URL building as the filters are dynamic and can create a lot of query parameters.

**Review by alessandrojean — CHANGES_REQUESTED** [review]

File: src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt (hunk 1 above)
Comment (dejavui) [reply src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt:160]: ext iutruyentranh search manga is the same ext truyentranh8

File: src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt [hunk 4]
```diff
@@ -1,85 +1,234 @@
...
+    val xem = "sort=xem"
+    val chap = "sort=chap"
+    val act = "act=search"
+    val andor = "andor=and"
+    val pages = "page="
 
     override fun popularMangaRequest(page: Int): Request {
-        return GET("$baseUrl/genre/$page?popular", headers)
+        return GET("$baseUrl/$search?$act&$andor&$xem&$view&$pages$page", headers)
     }
```
Comment (alessandrojean) [inline src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt:44]: Literally no need at all to do that... Just do like this

```kotlin
return GET("$baseUrl/search.php?act=search&andor=and&sort=xem&view=thumb&page=$page", headers)
```

Same to other similar instances.

File: src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt (hunk 1 above)
Comment (alessandrojean) [reply src/vi/iutruyentranh/src/eu/kanade/tachiyomi/extension/vi/iutruyentranh/IuTruyenTranh.kt:160]: Being the same doesn't mean it's correct.

**Review by alessandrojean — CHANGES_REQUESTED** [review]
There's also a build error, please fix it.

**Pull request closed without merging**